src/api/posts/get/getPosts.js

```diff
@@ -1,15 +1,25 @@
+//imports
 import axios, * as others from 'axios'
+import { ref } from 'vue'
 
+//data
+export const loadingData = ref(false)
+export const preLoaderStatus = ref(false)
+
+//methods
 export async function getAllPosts () {
   const SERVER_URL = 'http://localhost:3000/posts'
-
+  preLoaderStatus.value = true
```

Comment:
I suppose somethere in UI component you have a preloader layout...maybe it is a better idea to change flag above the method call and then change back to false, for example:

let pleloaderFlag =  false
let noDataComponent = false

....

preloaderFlag = true
const requestResult = await myAsyncMethod()
preloaderFlag = false

if (requestResult ) {
 //do something
}  else {
noDataComponent = true
// static component renders in v-if='noDataComponent' in template block
}

PLUS: this method of decompostion can help you to get rid of stateful variables in async methods in some cases